core/invoker/src/main/scala/whisk/core/containerpool/ContainerPool.scala

```diff
@@ -120,11 +120,7 @@ class ContainerPool(
         // Container got removed
         case ContainerRemoved =>
             freePool.remove(sender())
-            busyPool.remove(sender())
-
-        // Activation completed
-        case ActivationCompleted =>
-            feed ! ContainerReleased
+            busyPool.remove(sender()).foreach(_ => feed ! ContainerReleased)
```

Comment:
Correct. In effect this should be entirely the same as before but with less boilerplate and it actually sends the message **after** we're free again. With correct buffering upfront this completely removes the "self-send" tight-loop, which is desirable.